var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -295,6 +323,41 @@ def cmake_args(self):
                 '-DBZIP2_LIBRARIES=%s' % spec['bzip2'].libs.joined(';')
             ])
 
+        # Doxygen documentation
+        options.extend([
+            '-DDEAL_II_COMPONENT_DOCUMENTATION=%s' %
+            ('ON' if '+doc' in spec else 'OFF'),
+        ])
+
+        # Examples / tutorial programs
+        options.extend([
+            '-DDEAL_II_COMPONENT_EXAMPLES=%s' %
+            ('ON' if '+examples' in spec else 'OFF'),
+        ])
+
+        # Deal with C++ standard
+        if spec.variants['cxxstd'].value != 'default':
+            cxxstd = spec.variants['cxxstd'].value
+
+            if spec.satisfies('^cmake@3.1:'):
+                options.extend([
+                    '-DCMAKE_CXX_STANDARD={0}'.format(cxxstd)
+                ])
+            else:
+                flag = getattr(self.compiler, 'cxx{0}_flag'.format(cxxstd))
+                if flag:
+                    cxx_flags.append(flag)
```

Comment:
I need to come back at a later stage and check if these flags should be substituted for `options.extend(['-DDEAL_II_WITH_CXX{0}:BOOL=ON'.format(cxxstd)])`. I don't think that this would work in the case of specifying `2a`, but I'll see if I can make this work properly too. I seem to remember this working, but I should verify it again.